File: Java/time-conversion.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Solution {

    static String timeConversion(String s) throws ParseException{
        
        SimpleDateFormat output = new SimpleDateFormat("HH:mm:ss");
        SimpleDateFormat input = new SimpleDateFormat("hh:mm:ssaa");
        
        Date time = input.parse(s);
        return output.format(time);
        
    }

    private static final Scanner scan = new Scanner(System.in);

    public static void main(String[] args) throws IOException,ParseException {
        BufferedWriter bw = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = scan.nextLine();

        String result = timeConversion(s);

        bw.write(result);
        bw.newLine();

        bw.close();
    }
}

File: Python/time-conversion.py
#!/bin/python3

import os
import sys


def timeConversion(s):
    am_pm = s[-2:];
    s = s.replace(am_pm,'').split(':')
    if am_pm == "PM":
        s[0] = str(24-abs(int(s[0])-12))
    s[0] = '00' if s[0] == '12' and am_pm == 'AM' else s[0]
    s[0] = '12' if s[0] == '24' and am_pm == 'PM' else s[0]
    return ':'.join(s)

if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    f.write(result + '\n')

    f.close()
